### PR TITLE
Add Dubbo Response Code Mapping

### PR DESCRIPTION
**Pull request by rickey17**
### Issues associated with this PR

Dubbo Response Code Mapping is missing.

### Solutions
Add and Registry Mapping

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/protocol/xprotocol/dubbo/mapping.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+type dubboStatusMapping struct{}
+
+func (m *dubboStatusMapping) MappingHeaderStatusCode(ctx context.Context, headers types.HeaderMap) (int, error) {
+	cmd, ok := headers.(xprotocol.XRespFrame)
+	if !ok {
+		return 0, errors.New("no response status in headers")
+	}
+	code := uint16(cmd.GetStatusCode())
+	switch code {
+	case 20:
```
Comment (champly) [inline pkg/protocol/xprotocol/dubbo/mapping.go:24]: You can use [status](https://github.com/mosn/mosn/blob/b2d8718b12ab1f57f09d85042c24b6bd09b8cd19/pkg/protocol/xprotocol/dubbo/types.go#L63) replace number

Comment (rickey17) [reply pkg/protocol/xprotocol/dubbo/mapping.go]: done

Comment (champly) [reply pkg/protocol/xprotocol/dubbo/mapping.go]: include unit test, use `pkg/types/constant.go` const

File: pkg/protocol/xprotocol/dubbo/types.go [hunk 2]
```diff
@@ -46,6 +46,32 @@ const (
...
+	RespStatusServerTimeout                  = 31
+	RespStatusBadRequest                     = 40
+	RespStatusBadResponse                    = 50
+	RespStatusServiceNotFound                = 60
+	RespStatusServiceError                   = 70
+	RespStatusServerError                    = 80
+	RespStatusClientError                    = 90
+	RespStatusServerThreadpoolExhaustedError = 100
+)
+
```
Comment (champly) [inline pkg/protocol/xprotocol/dubbo/types.go:74]: you can pull the latest master

Comment (rickey17) [reply pkg/protocol/xprotocol/dubbo/types.go:74]: I need to use the develop branch

Comment (champly) [reply pkg/protocol/xprotocol/dubbo/types.go:74]: The latest master remove codecov test.

File: pkg/protocol/xprotocol/dubbo/mapping_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	frame := &Frame{
+		Header: Header{
+			Status: 0,
+		},
+	}
+	code, err = dubboStatusMapping.MappingHeaderStatusCode(nil, frame)
+	equal(t, code, 500)
+	equal(t, err == nil, true)
+
+	frame.Header.Status = 20
```
Comment (champly) [inline pkg/protocol/xprotocol/dubbo/mapping_test.go:23]: Use const varibale.

**Comment by nejisama** [comment]
Add unit-test.

File: pkg/protocol/xprotocol/dubbo/mapping.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return http.StatusOK, nil
+	case 30, 31:
+		return http.StatusGatewayTimeout, nil
```
Comment (zonghaishang) [inline pkg/protocol/xprotocol/dubbo/mapping.go:27]: dubbo framework status code definition:  https://github.com/apache/dubbo/blob/master/dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/Response.java

    /**
     * clien side timeout.
     */
    public static final byte CLIENT_TIMEOUT = 30;

    /**
     * server side timeout.
     */
    public static final byte SERVER_TIMEOUT = 31;


You must ensure that all status codes do not break the original semantics

**Review by zonghaishang — CHANGES_REQUESTED** [review]

**Comment by nejisama** [comment]
outdated pull request. if you still need this changed, make a new one

**Pull request closed without merging**